tests/dopamine/replay_memory/circular_replay_buffer_test.py

```diff
@@ -638,6 +637,50 @@ def testLoad(self):
     self.assertEqual(memory.add_count, self._test_add_count)
     self.assertAllClose(memory.invalid_range, self._test_invalid_range)
 
+  @staticmethod
+  def _create_locked_memory():
+    memory = circular_replay_buffer.OutOfGraphReplayBuffer(
+        observation_shape=(2,),
+        stack_size=1,
+        replay_capacity=10,
+        batch_size=2)
+    return memory
+
+  def test_memory_locks_itself(self):
+    """Tests that adding an element to the buffer blocks the lock."""
+    memory = self._create_locked_memory()
+    memory._lock = mock.Mock()
+    memory._lock.__enter__ = mock.Mock()
+    memory._lock.__exit__ = mock.Mock()
+
+    # Check that buffer is empty at first.
+    self.assertEqual(memory.add_count, 0)
+    # Add one element.
+    memory.add((1, 2), 0, 0, False)
+    # Check that buffer contains one element.
+    self.assertEqual(memory.add_count, 1)
+    # Check that the lock went throughh the proper lock/unlock process.
+    memory._lock.__enter__.assert_called_once()
+    memory._lock.__exit__.assert_called_once()
+
+  def test_memory_is_locked(self):
+    """Tests that when the lock is blocked elements cannot be added."""
```

Comment:
When you get to this, could you please ping me?  This test is hard for me to think about -- I don't know what the desired behavior under test is.

Reply:
I got to this - please ping me when you are online.